.werft/util/shell.ts

```diff
@@ -62,6 +62,66 @@ export function exec(cmd: string, options?: ExecOptions): ChildProcess | shell.S
     }
 }
 
+/**
+ * Execute a command and stream output logs.
+ *
+ * If a slice is given logs are streamed using the werft log syntax; else they're streamed directly
+ * to stderr/stdout.
+ */
+export async function execStream(command: string, options: ExecOptions ): Promise<ExecResult> {
+    const werft = getGlobalWerftInstance();
+
+    options = options || {};
+
+    if (options.slice) {
+        options.silent = true;
+    }
+
+    const child = shell.exec(command, {...options, async: true});
+
+    let stdout = '';
+    let stderr = '';
+
+    // note: the stdout/stderr event handlers aren't guaranteed to receive buffers that are always
+    // newline terminated. The original log messages can be preserved by finding the index of the
+    // last newline, printing up until that newline, buffering the remaining message, appending
+    // to that buffer on the next call, and finally flushing the buffers then the process exits.
```

Comment:
I think we'd get nicer output in Werft if we did decide to go with the solution outlined in your comment. What made you decide not to? ☺️

Reply:
In short: doing it right is a bit complex, and I'm really low on time right now.

In long: The self-hosted test and preview environments run a lot of very long, very slow commands, so it can be upwards of 25 minutes to determine if a single step failed. Earlier on we were having issues with EKS tests where nodes would fail to attach - but the failure would happen 5 minutes in but you'd only find out after Terraform timed out after 30 minutes. The absence of streamed output is a major drain on velocity.

With behavior in this PR, you end up with slightly weird command output that's sprinkled with a few newlines; it's a minor irritation but pretty tolerable. Fixing this requires a bit of time and patience; it's doable but it's not critical. Since the self-hosted installer tests are the only systems that are impacted by this change we're not going to mangle the test output for anyone that won't expect it. I'd like to ship this as a skateboard, immediately reap the benefits of the improvements, and continue iterating on command execution quality of life fixups.

To emphasize, this is a _massive_ quality of life improvement for the self hosted tests. I don't like merging code that has known quirks like this but I'm painfully low on time right now. I'd like to spend a goodwill token on this if at all possible; 😇 incorporating this fix will really improve velocity in the pipelines and represents a small, well defined amount of technical debt.